src/ast/dml.rs

```diff
@@ -124,6 +124,7 @@ pub struct CreateTable {
     pub default_charset: Option<String>,
     pub collation: Option<String>,
     pub on_commit: Option<OnCommit>,
+    pub with_order: Vec<Vec<OrderByExpr>>,
```

Comment:
Could we include here a link to the docs where this syntax comes from? (e.g. similar to the on_cluster below)

Reply:
FWIW I did some research and could not find any existing databases that use this syntax.

ClickHouse does seem to have a way to specify order as part of a `CREATE TABLE` statement: https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree

@mertak-synnada rather than introducing special DataFusion only syntax support, what do you think about extending DataFusion to use the existing ClickHouse syntax?

```sql
CREATE TABLE ....
ORDER BY <...>
```

We might have to change the `GenericDialect` or add some feature to the `Dialect` trait to permit other dialects to parse such syntax, but it would be nice to align to some other existing syntax rather than creating something new

Reply:
I see two possible ways forward:
1. We can change DF to support `ORDER BY` after `CREATE` and deprecate `WITH ORDER`. This way we would have Clickhouse-like syntax for both memory and external tables, and `WITH ORDER` syntax only for external tables. The latter asymmetry is a little weird but we can explain it away by deprecating `WITH ORDER`.
2. We can support both `ORDER BY` and `WITH ORDER` for both memory and external tables in DF (they would be synonyms). We'd need to add `WITH ORDER` for ordinary `CREATE`s here though.

I am OK with both, with a slight preference to 2 because DF has had `WITH ORDER` for a while now. What do you think?

Reply:
I think in this case we should distinguish between DataFusion's needs and sqlparser-rs's needs

If we are going to introduce `WITH ORDER` to sqlparser-rs I think it should follow the existing pattern of being connected to a specific dialect (in this case `DatafusionDialect`). 

Thus I would personally vote for option 1 (change DF)

> 1. We can change DF to support ORDER BY after CREATE and deprecate WITH ORDER. This way we would have Clickhouse-like syntax for both memory and external tables, and WITH ORDER syntax only for external tables. The latter asymmetry is a little weird but we can explain it away by deprecating WITH ORDER.

Reply:
Makes sense - let's do it